src/shogun/base/SGObject.h

```diff
@@ -421,18 +417,9 @@ namespace shogun
 			    value, "Cannot add to %s::%s, no object provided.\n",
 			    get_name(), name.c_str());
 			Tag<std::vector<std::shared_ptr<T>>> tag_vector(name);
-
-			auto push_back_lambda = [&value, this](auto& array) {
-				array.push_back(value);
-			};
-
-			if (sgo_details::dispatch_array_type<T>(
-			        shared_from_this(), name, std::move(push_back_lambda)))
-				return;
-			SG_ERROR(
-			    "Cannot add object %s to array parameter %s::%s of type %s.\n",
-			    value->get_name(), get_name(), name.c_str(),
-			    demangled_type<T>().c_str());
+			auto dispatched = get(tag_vector);
+			dispatched.push_back(value);
+			update_parameter(BaseTag(name), make_any(dispatched), false);
```

Comment:
@karlnapf we can't get a any_cast value by reference, we just get a copy. So need to put the copy to replace the original after pushing the value. But some parameters are readonly, so I added a flag to skip checks... It's not great but is the simplest solution until we can come up with a better design to update Any objects...

Reply:
Ok